### PR TITLE
RHOS Deploy Fix

### PR DESCRIPTION
**Pull request by chase-kiefer**
Add missing SCC and other supporting configurations so deployment to RHOS works as expected.

- Add nginx-mesh-nats-permissions SecurityContextConstraints
- Add creation of supporting ClusterRole and RoleBinding
- Set allowHostDirVolumePlugin to true
- Create empty spire-data
- Reduce volumes in mesh-api SCC

### Proposed changes
Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue here in this description (not in the title of the PR).

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-service-mesh/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: helm-chart/templates/nats.yaml [hunk 1]
```diff
@@ -16,6 +16,75 @@ metadata:
...
+seLinuxContext:
+  type: MustRunAs
+supplementalGroups:
+  type: MustRunAs
+users: []
+volumes:
+  - secret
+  - csi
+  - configMap
+  - emptyDir
```
Comment (sjberman) [inline helm-chart/templates/nats.yaml:55]: nit: move all list indentations to the left for consistency (nested directly under the parent) for all resources you've added

File: helm-chart/templates/nginx-mesh-api.yaml [hunk 2]
```diff
@@ -387,10 +387,6 @@ runAsUser:
   type: MustRunAsNonRoot
 fsGroup:
   type: MustRunAs
-volumes:
-- configMap
-- csi
-- secret
```
Comment (sjberman) [inline helm-chart/templates/nginx-mesh-api.yaml:393]: There's another SCC at the bottom of this file that likely needs attention too, unless you already checked it out.

Comment (sjberman) [reply helm-chart/templates/nginx-mesh-api.yaml:393]: I actually can't remember what that last SCC is for...it's bound to the system authenticated group

Comment (chase-kiefer) [reply helm-chart/templates/nginx-mesh-api.yaml:393]: I was able to remove most of the volumes from that SCC and still deploy. The remaining are required for the API pod to come up.

Comment (sjberman) [reply helm-chart/templates/nginx-mesh-api.yaml:393]: Cool. I'm not sure why we need the two SCCs or how they differ...worth looking into another time probably.

File: helm-chart/templates/nats.yaml [hunk 3]
```diff
@@ -16,6 +16,71 @@ metadata:
...
+allowedCapabilities:
+- NET_ADMIN
+- KILL
+defaultAddCapabilities: null
+fsGroup:
+  type: MustRunAs
+priority: null
+readOnlyRootFilesystem: false
+requiredDropCapabilities:
+  - ALL
```
Comment (sjberman) [inline helm-chart/templates/nats.yaml:42]: nit: one missed indentation here

File: helm-chart/templates/nats.yaml [hunk 4]
```diff
@@ -16,6 +16,71 @@ metadata:
...
+kind: ClusterRole
+metadata:
+  name: system:openshift:scc:nginx-mesh-nats-permissions
+  labels:
+    app.kubernetes.io/part-of: nginx-service-mesh
+rules:
+  - apiGroups: ["security.openshift.io"]
+    resources: ["securitycontextconstraints"]
+    resourceNames: ["nginx-mesh-nats-permissions"]
+    verbs: ["use"]
```
Comment (sjberman) [inline helm-chart/templates/nats.yaml:67]: and here

File: helm-chart/templates/nats.yaml [hunk 5]
```diff
@@ -16,6 +16,71 @@ metadata:
...
+  labels:
+    app.kubernetes.io/part-of: nginx-service-mesh
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:openshift:scc:nginx-mesh-nats-permissions
+subjects:
+  - kind: ServiceAccount
+    name: nats
+    namespace: {{ .Release.Namespace }}
```
Comment (sjberman) [inline helm-chart/templates/nats.yaml:82]: and here

File: helm-chart/templates/nats.yaml [hunk 6]
```diff
@@ -16,6 +16,71 @@ metadata:
     app.kubernetes.io/part-of: nginx-service-mesh
 data:
   nats.conf: {{ .Files.Get "configs/nats.conf" | quote }}
+{{- if eq .Values.environment "openshift" }}
+---
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
+  name: nginx-mesh-nats-permissions
```
Comment (sjberman) [inline helm-chart/templates/nats.yaml:24]: be sure to add the label for `part-of: nsm`

**Review by sjberman — APPROVED** [review]

**Review by JoeMcGuire98 — APPROVED** [review]

**Pull request merged**